src/main/java/org/spongepowered/api/world/storage/WorldProperties.java

```diff
@@ -0,0 +1,352 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.world.storage;
+
+import org.spongepowered.api.service.persistence.data.DataQuery;
+
+import org.spongepowered.api.service.persistence.data.DataView;
+import com.google.common.base.Optional;
+import com.flowpowered.math.vector.Vector3i;
+import org.spongepowered.api.entity.player.gamemode.GameMode;
+import org.spongepowered.api.service.persistence.DataSerializable;
+import org.spongepowered.api.service.persistence.data.DataContainer;
+import org.spongepowered.api.world.DimensionType;
+import org.spongepowered.api.world.WorldBorder;
+import org.spongepowered.api.world.GeneratorType;
+import org.spongepowered.api.world.difficulty.Difficulty;
+
+import java.util.Map;
+
+/**
+ * Represents the WorldProperties which are persisted across runtime instances.
+ */
+public interface WorldProperties extends DataSerializable {
+
+    /**
+     * Gets the default spawn position of this world.
+     * 
+     * @return The spawn position
+     */
+    Vector3i getSpawnPosition();
+
+    /**
+     * Sets the default spawn position of this world.
+     * 
+     * @param position The spawn position
+     */
+    void setSpawnPosition(Vector3i position);
+
+    /**
+     * Gets the type of the generator for this world.
+     * 
+     * @return The type
+     */
+    GeneratorType getType();
+
+    /**
+     * Sets the type of this world.
+     * 
+     * @param type The type
+     */
+    void setType(GeneratorType type);
```

Comment:
`WorldBuilder.settings(DataContainer settings)`

See also: `GeneratorType.getGeneratorSettings()` a variant of which is what the builder will be expecting.